Sources/Confidence/Confidence.swift

```diff
@@ -1,45 +1,70 @@
 import Foundation
 
 public class Confidence: ConfidenceEventSender {
-    public var context: ConfidenceStruct
+    private let parent: ConfidenceContextProvider?
+    private var context: ConfidenceStruct
     public let clientSecret: String
     public var timeout: TimeInterval
     public var region: ConfidenceRegion
     public var initializationStrategy: InitializationStrategy
+    private var removedContextKeys: Set<String> = Set()
 
-    init(
+    required public init(
         clientSecret: String,
         timeout: TimeInterval,
         region: ConfidenceRegion,
-        initializationStrategy: InitializationStrategy
+        initializationStrategy: InitializationStrategy,
+        context: ConfidenceStruct = [:],
+        parent: ConfidenceEventSender? = nil
     ) {
-        self.context = [:]
         self.clientSecret = clientSecret
         self.timeout = timeout
         self.region = region
         self.initializationStrategy = initializationStrategy
+        self.context = context
+        self.parent = parent
     }
 
     // TODO: Implement actual event uploading to the backend
     public func send(definition: String, payload: ConfidenceStruct) {
         print("Sending: \"\(definition)\".\nMessage: \(payload)\nContext: \(context)")
     }
 
+
+    public func getContext() -> ConfidenceStruct {
+        let parentContext = parent?.getContext() ?? [:]
+        var reconciledCtx = parentContext.filter {
+            !removedContextKeys.contains($0.key)
+        }
+        self.context.forEach { entry in
+            reconciledCtx.updateValue(entry.value, forKey: entry.key)
+        }
+        return reconciledCtx
+    }
+
     public func updateContextEntry(key: String, value: ConfidenceValue) {
         context[key] = value
+        removedContextKeys.remove(key)
```

Comment:
i believe we don't need this since [here](https://github.com/spotify/confidence-openfeature-provider-swift/pull/89/files#diff-441da54f0ca3b493090d67c3fa56c1ae79a93a1f3257b7cff88ab9d55ebf8522R40) we will have the context key added again.